### PR TITLE
Nano: Troubshooting guide for BigDL-Nano

### PR DESCRIPTION
**Pull request by rnwang04**
## Description

As more and more new parameters added into BigDL-Nano, and more and more issues found during oob, it's necessary to record these issues so that if users meet the same issue, they can quickly find the reason and know how to solve it.

### 1. Why the change?

As more and more new parameters added into BigDL-Nano, and more and more issues found during oob, it's necessary to record these issues so that if users meet the same issue, they can quickly find the reason and know how to solve it.

### 2. User API changes

No change.

### 3. Summary of the change 

- New troubshooting guide

### 4. How to test?
- [x] Document test
https://ruonantetdoc.readthedocs.io/en/troubshooting/doc/Nano/Overview/troubshooting.html

### CODE REVIEW

File: docs/readthedocs/source/doc/Nano/Overview/troubshooting.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+## Installation
+### Why I fail to install openvino==2022.2 when ``pip install bgdl-nano[inference]``?
+
+Please check your system first as openvino 2022.x does not support centos anymore. Refer [OpenVINO release notes](https://www.intel.com/content/www/us/en/developer/articles/release-notes/openvino-relnotes-2021.html) for more details.
+
+## Inference
+
+### ``could not create a primitive descriptor iterator`` when using bf16 related methods
+
+Please make sure you use context manager provided by ``InferenceOptimizer.get_context``, you can refer this [howto guide for context manager]() for more details.
```
Comment (MeouSker77) [inline docs/readthedocs/source/doc/Nano/Overview/troubshooting.md:14]: no link to 'howto guide for context manager' ? Maybe we should add that howto guide before merging this PR

File: docs/readthedocs/source/doc/Nano/Overview/troubshooting.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+If you accelerate the model with ``accelerator=None`` by ``InferenceOptimizer.trace``/``InferenceOptimizer.quantize`` or it's just a normal torch.nn.Module, you have to pass original FP32 model to load pytorch model by ``InferenceOptimizer.load(dir, model=model)``.
+
+### Why my bf16 model is slower than fp32 model?
+
+You can first check whether your machine supports the bf16 instruction set first by ``lscpu | grep "avx512"``. If there is no ``avx512_bf16`` in the output, then, without instruction set support, the performance of bf16 cannot be guaranteed, and generally, its performance will deteriorate.
+
+### ``INVALID_ARGUMENT : Got invalid dimensions for input`` or ``[ PARAMETER_MISMATCH ] Can not clone with new dims.`` when do inference with OpenVINO / ONNXRuntime accelerated model
+
+This error usually occurs when the input of ``forward`` varies during inference, and such case needs you to manually set ``dynamic_axes`` parameter and pass ``dynamic_axes`` to ``trace/quantize``.
```
Comment (MeouSker77) [inline docs/readthedocs/source/doc/Nano/Overview/troubshooting.md:48]: may change **the input of ``forward``** to **the shape of ``forward``'s input**

**Review by MeouSker77 — APPROVED** [review]
LGTM

File: docs/readthedocs/source/doc/Nano/Overview/troubshooting.md (hunk 1 above)
Comment (rnwang04) [reply docs/readthedocs/source/doc/Nano/Overview/troubshooting.md:14]: sure, will merge that PR first.

File: docs/readthedocs/source/doc/Nano/Overview/troubshooting.md (hunk 2 above)
Comment (rnwang04) [reply docs/readthedocs/source/doc/Nano/Overview/troubshooting.md:48]: Would it be better : This error usually occurs when your dataset has dynamic input shape, and for such case needs you to manually set ``dynamic_axes`` parameter and pass ``dynamic_axes`` to ``trace/quantize``.

Comment (MeouSker77) [reply docs/readthedocs/source/doc/Nano/Overview/troubshooting.md:48]: > This error usually occurs when your dataset has dynamic input shape
when your dataset produces data with dynamic shape

File: docs/readthedocs/source/doc/Nano/Overview/troubshooting.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+Please check the shape of your input data first, we don't support ``channels_last`` for 3D input now. If your model is a 3D model or your input data is not 4D Tensor, normally ``channels_last`` option will fail.
+
+### Why ``InferenceOptimizer.load(dir)`` fails to load my model saved by ``InferenceOptimizer.save(model, dir)``
+
+If you accelerate the model with ``accelerator=None`` by ``InferenceOptimizer.trace``/``InferenceOptimizer.quantize`` or it's just a normal torch.nn.Module, you have to pass original FP32 model to load pytorch model by ``InferenceOptimizer.load(dir, model=model)``.
+
+### Why my bf16 model is slower than fp32 model?
+
+You can first check whether your machine supports the bf16 instruction set first by ``lscpu | grep "avx512"``. If there is no ``avx512_bf16`` in the output, then, without instruction set support, the performance of bf16 cannot be guaranteed, and generally, its performance will deteriorate.
```
Comment (TheaperDeng) [inline docs/readthedocs/source/doc/Nano/Overview/troubshooting.md:44]: You can first check whether your machine supports the bf16 instruction set first by lscpu | grep “bf16". If there is no avx512_bf16 or amx_bf16 in the output, then, without instruction set support, the performance of bf16 cannot be guaranteed, and generally, its performance will deteriorate.

**Review by TheaperDeng — COMMENTED** [review]
others LGTM, we could merge this PR first and add more issues and better organization in next PR.

**Comment by TheaperDeng** [comment]
merge this PR first and I will make corresponding changing according to the cmt. just to keep things rolling

**Pull request merged**